Backend/Controllers/ProductController.cs

```diff
@@ -0,0 +1,229 @@
+                }
+            }
+
+            return Problem();
+        }
+
+    
+        [HttpGet("{id}")]
+        public async Task<ActionResult<Product>> GetProduct(string id)
+        {
+            var product = await db.Products.QueryOne(p => p.Id == id);
+
+            if (product == null)
+            {
+                return NotFound();
+            }
+
+            return Ok(product);
+        }
+
+   
+        [HttpPatch("{id}")]
+        public async Task<IActionResult> UpdateProduct([FromBody] Product patchDoc)
+        {
+            var product = await db.Products.QueryOne(p => p.Id == patchDoc.Id);
+
+            if (product == null)
+            {
+                return NotFound();
+            }
+
+
+            if (patchDoc.Name != null)
+            {
+                product.Name = patchDoc.Name;
+            }
+
+            if (patchDoc.Amount != null)
+            {
+                product.Amount = patchDoc.Amount;
+            }
+
+            if (patchDoc.Price != null)
+            {
+                product.Price = patchDoc.Price;
+            }
+
+            if (patchDoc.PreviewImage != null)
+            {
+                product.PreviewImage = patchDoc.PreviewImage;
+            }
+
+            if (patchDoc.VideoUrl != null)
+            {
+                product.VideoUrl = patchDoc.VideoUrl;
+            }
+
+            if (patchDoc.SeoTitle != null)
+            {
+                product.SeoTitle = patchDoc.SeoTitle;
+            }
+
+            if (patchDoc.SeoSlug != null)
+            {
+                product.SeoSlug = patchDoc.SeoSlug;
+            }
+
+            if (patchDoc.SeoDescription != null)
+            {
+                product.SeoDescription = patchDoc.SeoDescription;
+            }
+
+            var saved = await db.Save();
+
+            return saved ? Ok(product) : Problem();
+        }
+
+
+        [HttpPut("{id}/publish")]
+        public async Task<IActionResult> PublishProduct(string id)
+        {
+            var product = await db.Products.QueryOne(p => p.Id == id);
+
+            if (product == null)
+            {
+                return NotFound();
+            }
+
+            product.IsDraft = false;
+
+            var saved = await db.Save();
+            return saved ? Ok(product) : Problem();
+        }
+
+
+        [HttpPost("{id}/images")]
+        public async Task<IActionResult> UploadProductImages(string id, [FromForm] ICollection<IFormFile> imageFiles)
+        {
+            if (imageFiles != null)
+            {
+                var product = await db.Products.QueryOne(p => p.Id == id);
+
+                if (product == null)
+                {
+                    return NotFound();
+                }
+
+                var productImagesToDelete = await db.ProductImages.QueryMany(pi => pi.ProductId == id);
+                db.ProductImages.RemoveRange(productImagesToDelete);
+                await imageService.SafeDelete(productImagesToDelete);
```

Comment:
no! you don't delete previous images. We should add by 1 image and delete by 1 image